process_report/tests/unit/invoices/test_pi_specific_invoice.py

```diff
@@ -1,68 +1,147 @@
+
+        pi_inv = test_utils.new_pi_specific_invoice(data=test_invoice)
+        output_invoice = pi_inv._get_pi_dataframe(test_invoice, "PI1")
+        self.assertTrue(answer_invoice_pi1.equals(output_invoice))
+
+        output_invoice = pi_inv._get_pi_dataframe(test_invoice, "PI2")
+        self.assertTrue(answer_invoice_pi2.equals(output_invoice))
 
-        pi_df = pandas.read_csv(output_dir.name + "/" + pi_csv_1)
-        self.assertEqual(len(pi_df["Manager (PI)"].unique()), 1)
-        self.assertEqual(
-            pi_df["Manager (PI)"].unique()[0], self.dataframe["Manager (PI)"][0]
+    @mock.patch("process_report.invoices.invoice.Invoice._filter_columns")
+    @mock.patch("os.path.exists")
+    @mock.patch("subprocess.run")
+    def test_export_pi(self, mock_subprocess_run, mock_path_exists, mock_filter_cols):
+        invoice_month = "2024-10"
+        test_invoice = self._get_test_invoice(
+            ["PI1", "PI1", "PI2", "PI2"],
+            [
+                "BU",
+                "BU",
+                "HU",
+                "HU",
+            ],
+            [100, 200, 300, 400],
+            group_name=[None, "G1", None, None],
         )
 
-        self.assertIn("ProjectA", pi_df["Project - Allocation"].tolist())
-        self.assertIn("ProjectB", pi_df["Project - Allocation"].tolist())
-        self.assertIn("ProjectC", pi_df["Project - Allocation"].tolist())
+        mock_filter_cols.return_value = test_invoice
+        mock_path_exists.return_value = True
+        output_dir = tempfile.TemporaryDirectory()
 
-        pi_df = pandas.read_csv(output_dir.name + "/" + pi_csv_2)
-        self.assertEqual(len(pi_df["Manager (PI)"].unique()), 1)
-        self.assertEqual(
-            pi_df["Manager (PI)"].unique()[0], self.dataframe["Manager (PI)"][3]
+        pi_inv = test_utils.new_pi_specific_invoice(
+            output_dir.name, invoice_month, data=test_invoice
         )
+        pi_inv.process()
+        pi_inv.export()
+        pi_pdf_1 = f"{output_dir.name}/BU_PI1_{invoice_month}.pdf"
+        pi_pdf_2 = f"{output_dir.name}/HU_PI2_{invoice_month}.pdf"
 
-        self.assertIn("ProjectD", pi_df["Project - Allocation"].tolist())
-        self.assertIn("ProjectE", pi_df["Project - Allocation"].tolist())
-        self.assertNotIn("ProjectA", pi_df["Project - Allocation"].tolist())
-        self.assertNotIn("ProjectB", pi_df["Project - Allocation"].tolist())
-        self.assertNotIn("ProjectC", pi_df["Project - Allocation"].tolist())
+        for i, pi_pdf_path in enumerate([pi_pdf_1, pi_pdf_2]):
+            chrome_arglist, _ = mock_subprocess_run.call_args_list[i]
+            answer_arglist = [
+                "usr/bin/chromium",
+                "--headless",
+                "--no-sandbox",
+                f"--print-to-pdf={pi_pdf_path}",
+                "--no-pdf-header-footer",
+            ]
+            for answer_arg in answer_arglist:
+                self.assertTrue(answer_arg in chrome_arglist[0])
```

Comment:
I would be more constrained in your check:

```suggestion
            self.assertTrue(answer_arglist == chrome_arglist[0][:-1])
```